spec/play_store/subscription_purchases/subscription_purchase_spec.rb

```diff
@@ -109,6 +109,34 @@
     end
   end
 
+  describe "canceled subscription by user" do
+    let(:fake_subscription_purchase) do
+      FakeSubscriptionPurchase.new(
+        kind: "androidpublisher#subscriptionPurchase",
+        start_time_millis: 1459540113244,
+        expiry_time_millis: 1462132088610,
+        user_cancellation_time_millis: 1461872888000,
+        auto_renewing: true,
+        developer_payload: "payload that gets stored and returned",
+        cancel_reason: 0,
+        payment_state: 1,
+      )
+    end
+
+    it "is canceled?" do
+      subject.canceled_by_user?.must_be_true
+    end
+
+    it "returns the user_cancellation_time_millis" do
+      subject.user_cancellation_time_millis.must_equal 146_187_288_800_0
```

Comment:
I would suggest starting the integer grouping from the right. That trailing zero looks strange :-) 
```suggestion
      subject.user_cancellation_time_millis.must_equal 1_461_872_888_000
```